eslint/babel-eslint-parser/src/convert/convertAST.js

```diff
@@ -82,20 +83,16 @@ function convertNodes(ast, code) {
   const state = { source: code };
 
   // Monkey patch visitor keys in order to be able to traverse the estree nodes
-  t.VISITOR_KEYS.Property = t.VISITOR_KEYS.ObjectProperty;
-  t.VISITOR_KEYS.MethodDefinition = [
-    "key",
-    "value",
-    "decorators",
-    "returnType",
-    "typeParameters",
-  ];
+  t.VISITOR_KEYS.ChainExpression = VISITOR_KEYS.ChainExpression;
+  t.VISITOR_KEYS.Property = VISITOR_KEYS.Property;
+  t.VISITOR_KEYS.MethodDefinition = VISITOR_KEYS.MethodDefinition;
 
   traverse(ast, astTransformVisitor, null, state);
 
   // These can be safely deleted because they are not defined in the original visitor keys.
-  delete t.VISITOR_KEYS.Property;
+  delete t.VISITOR_KEYS.ChainExpression;
   delete t.VISITOR_KEYS.MethodDefinition;
+  delete t.VISITOR_KEYS.Property;
```

Comment:
Unrelated, but the monkey-patching should be done in a `try/finally` in case `traverse()` throws.